pulsar/pulsar-impl/src/main/java/com/datastax/oss/sink/pulsar/PulsarRecordProcessor.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.sink.pulsar;
+
+import com.datastax.oss.sink.EngineAPIAdapter;
+import com.datastax.oss.sink.RecordProcessor;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PulsarRecordProcessor<Coat, Payload>
+    extends RecordProcessor<LocalRecord<Coat, Payload>, Header> {
```

Comment:
The `Coat` as the generic type name seems non-intuitive to me. Can we create a different name?

Reply:
yes, I'm thinking of renaming it to `Input`